go/cmd/dolt/commands/status.go

```diff
@@ -63,215 +96,574 @@ func (cmd StatusCmd) ArgParser() *argparser.ArgParser {
 
 // Exec executes the command
 func (cmd StatusCmd) Exec(ctx context.Context, commandStr string, args []string, dEnv *env.DoltEnv, cliCtx cli.CliContext) int {
+	// parse arguments
 	ap := cmd.ArgParser()
 	help, _ := cli.HelpAndUsagePrinters(cli.CommandDocsForCommandString(commandStr, statusDocs, ap))
 	apr := cli.ParseArgsOrDie(ap, args, help)
+	showIgnoredTables := apr.Contains(cli.ShowIgnoredFlag)
 
-	roots, err := dEnv.Roots(ctx)
+	// configure SQL engine
+	queryist, sqlCtx, closeFunc, err := cliCtx.QueryEngine(ctx)
 	if err != nil {
 		return handleStatusVErr(err)
 	}
+	if closeFunc != nil {
+		defer closeFunc()
+	}
 
-	staged, notStaged, err := diff.GetStagedUnstagedTableDeltas(ctx, roots)
+	// get status information from the database
+	pd, err := cmd.createPrintData(err, queryist, sqlCtx, showIgnoredTables)
 	if err != nil {
 		return handleStatusVErr(err)
 	}
 
-	ws, err := dEnv.WorkingSet(ctx)
+	// print everything
```

Comment:
This is not a useful comment. It's clear that's what happening by the function name